src/main/java/com/digitalasset/testing/ledger/SandboxCommunicator.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright 2019 Digital Asset (Switzerland) GmbH and/or its affiliates
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.digitalasset.testing.ledger;
+
+import com.daml.ledger.rxjava.DamlLedgerClient;
+import com.digitalasset.ledger.api.v1.LedgerIdentityServiceGrpc;
+import com.digitalasset.ledger.api.v1.LedgerIdentityServiceOuterClass;
+import com.digitalasset.ledger.api.v1.testing.TimeServiceGrpc;
+import com.digitalasset.testing.ledger.clock.SandboxTimeProvider;
+import com.digitalasset.testing.store.DefaultValueStore;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Consumer;
+
+import static com.digitalasset.testing.utils.SandboxUtils.getSandboxPort;
+import static com.digitalasset.testing.utils.SandboxUtils.waitForSandbox;
+
+public class SandboxCommunicator {
```

Comment:
This name is a bit odd. Would it make sense to call this `Sandbox`, and the class now called Sandbox could be renamed to `SandboxResource`?

Reply:
Yes this name is odd, we need to find it a better one. The renaming `Sandbox` -> `SandboxResource` is cool. My problem with calling SandboxCommunicator as Sandbox is that it contains ledger adapter, channel and client as well, I am not sure those are really parts of "Sandbox".